Sofa/framework/Type/src/sofa/type/RGBAColor.h

```diff
@@ -22,33 +22,57 @@
 #pragma once
 #include <sofa/type/config.h>
 
-#include <sofa/type/fixed_array.h>
-
 #include <ostream>
 #include <istream>
 #include <string>
 #include <cmath>
+#include <array>
+#include <algorithm>
+
+#include <sofa/type/fixed_array.h>
 
 namespace sofa::type
 {
 
-using sofa::type::fixed_array ;
-
-
 #define RGBACOLOR_EQUALITY_THRESHOLD 1e-6
 
 /**
  *  \brief encode a 4 RGBA component color
  */
-class SOFA_TYPE_API RGBAColor : public fixed_array<float, 4>
+class SOFA_TYPE_API RGBAColor
 {
+private:
+    static constexpr sofa::Size NumberOfComponents = 4;
+
+    using ComponentArray = std::array<float, NumberOfComponents>;
+
+    ComponentArray m_components;
+
 public:
-    static RGBAColor fromString(const std::string& str) ;
-    static RGBAColor fromFloat(float r, float g, float b, float a) ;
-    static RGBAColor fromVec4(const fixed_array<float, 4>& color) ;
-    static RGBAColor fromVec4(const fixed_array<double, 4>& color) ;
+    constexpr RGBAColor()
+        : m_components{ 1.f, 1.f, 1.f, 1.f } {}
+
+    constexpr RGBAColor(const std::array<float, NumberOfComponents>& c)
```

Comment:
```suggestion
    constexpr explicit RGBAColor(const std::array<float, NumberOfComponents>& c)
```

Reply:
I committed this suggestion, but reverted back as it breaks quite a lot of code mixing Vec4f/RGBAColor in the codebase. 😓
I would like this PR to be as less breakable as possible so for now....
After fixing the various mixups here and there, I will add the keyword explicit later